### PR TITLE
ghc: avoid build segfault with Xcode 7.3's nm

### PR DESCRIPTION
**Pull request by ilovezfs**
Reproduce the segfault before testing a fix for it.
Expect Mavericks and Yosemite build success, El Capitan build failure.

### CODE REVIEW

**Comment by ilovezfs** [comment]
Segfault reproduced as expected on 10.11:
```
make[1]: *** [compiler/stage2/dll-split.stamp] Segmentation fault: 11
```
http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/657/version=el_capitan/console

Next, force pushing with the `nm-classic` hack.

**Comment by ilovezfs** [comment]
Bug confirmed by upstream: https://ghc.haskell.org/trac/ghc/ticket/11951#comment:3
>This is a bug in llvm-nm; see #11744. It has been fixed upstream but Apple has yet to ship a fixed XCode release.

Apple broke POSIX compliance: https://ghc.haskell.org/trac/ghc/ticket/11744

The bug is also mentioned in 8.0.1rc3 release notes:
https://mail.haskell.org/pipermail/ghc-devs/2016-April/011862.html

Recommended workaround there is `--with-nm=nm-classic`. Hopefully, that will work with 7.10.3b as well.

**Comment by ilovezfs** [comment]
@mikemcquaid @xu-cheng I'm wondering if `nm` is going to need to be added to superenv at least temporarily since Apple broke POSIX compliance in Xcode 7.3 and who knows if or when they're going to fix it.

**Comment by mikemcquaid** [comment]
@ilovezfs Let's wait until it affects at least 3+ formula before we add to superenv.

**Comment by ilovezfs** [comment]
@mikemcquaid OK. Just a possible source of breakage to keep an eye out for.

**Comment by ilovezfs** [comment]
So it appears that `--with-nm=nm-classic` doesn't work with 7.10.3b.

@mikemcquaid @xu-cheng Do you prefer what's currently in the PR, or
```
    xcode_nm_classic = Pathname.new(MacOS.active_developer_dir)/"Toolchains/XcodeDefault.xctoolchain/usr/bin/nm-classic"
    clt_nm_classic = Pathname.new(MacOS.active_developer_dir)/"usr/bin/nm-classic"
    nm_classic = if xcode_nm_classic.exist?
      xcode_nm_classic
    elsif clt_nm_classic.exist?
      clt_nm_classic
    end
    args << "--with-nm=#{nm_classic}" unless nm_classic.nil?
```
or something else?

File: Formula/ghc.rb [hunk 1]
```diff
@@ -48,6 +49,24 @@ class Ghc < Formula
   end
 
   def install
+    # As of Xcode 7.3 (and the corresponding CLT) `nm` is a symlink to `llvm-nm`
+    # and the old `nm` is renamed `nm-classic`. Building with the new `nm`, a
+    # segfault occurs with the following error:
+    #   make[1]: * [compiler/stage2/dll-split.stamp] Segmentation fault: 11
+    # Reported 19th Apr 2016 here: https://ghc.haskell.org/trac/ghc/ticket/11951
+    if ENV.compiler == :clang && MacOS.clang_build_version >= 703
+      mkdir "brewtools"
```
Comment (DomT4) [inline Formula/ghc.rb]: FWIW you don't need to do this explicitly; `.write` below will create the path if it doesn't exist.

**Comment by mikemcquaid** [comment]
Other than @DomT4's comment 👍

File: Formula/ghc.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/ghc.rb]: Wow. I wish that were not true, but thanks for the tip.

Comment (DomT4) [inline Formula/ghc.rb]: Would this be a bad time to mention you could have merged it as-is and then fixed this? 😄. There's no material change to the build, so technically you could've avoided going back through CI.

**Comment by ilovezfs** [comment]
Timed out. Swell.

**Pull request closed without merging**